lib/components/src/layout/desktop.js

```diff
@@ -0,0 +1,302 @@
+    flexDirection: addonPanelInRight ? 'row' : 'column',
+    alignItems: 'stretch',
+    position: 'absolute',
+    width: '100%',
+    height: '100%',
+    padding: addonPanelInRight
+      ? `${layoutMargin} ${layoutMargin} ${layoutMargin} 0`
+      : `0px ${layoutMargin} ${layoutMargin} 0`,
+    boxSizing: 'border-box',
+  })
+);
+
+const resizerCursor = isVert => (isVert ? 'col-resize' : 'row-resize');
+
+const storiesResizerStyle = (showStoriesPanel, storiesPanelOnTop) => ({
+  cursor: showStoriesPanel ? resizerCursor(!storiesPanelOnTop) : undefined,
+  height: storiesPanelOnTop ? 10 : 'auto',
+  width: storiesPanelOnTop ? '100%' : 10,
+  zIndex: 1,
+});
+
+const addonResizerStyle = (showAddonPanel, addonPanelInRight) => ({
+  cursor: showAddonPanel ? resizerCursor(addonPanelInRight) : undefined,
+  height: addonPanelInRight ? '100%' : 10,
+  width: addonPanelInRight ? 10 : '100%',
+  zIndex: 1,
+});
+
+const ContentPanel = styled('div')(
+  ({ addonPanelInRight, storiesPanelOnTop, theme: { layoutMargin } }) => ({
+    position: 'absolute',
+    boxSizing: 'border-box',
+    width: '100%',
+    height: '100%',
+    padding: addonPanelInRight
+      ? `${layoutMargin} 0 ${layoutMargin} 0`
+      : `${layoutMargin} ${layoutMargin} 0 0`,
+    paddingTop: storiesPanelOnTop ? 0 : layoutMargin,
+  })
+);
+
+const PreviewWrapper = styled('div')(
+  ({ theme }) => ({
+    width: '100%',
+    height: '100%',
+    boxSizing: 'border-box',
+    overflow: 'auto',
+    WebkitOverflowScrolling: 'touch',
+    position: 'relative',
+    backgroundColor: theme.mainFill,
+  }),
+  ({ fullscreen, theme }) =>
+    fullscreen
+      ? {
+          position: 'fixed',
+          left: 0,
+          right: 0,
+          top: 0,
+          zIndex: 1,
+          border: 0,
+          margin: 0,
+          padding: 0,
+        }
+      : {
+          border: theme.mainBorder,
+          borderRadius: theme.mainBorderRadius,
+
+          display: 'flex',
+          justifyItems: 'center',
+          alignItems: 'center',
+          '& > *': {
+            margin: 'auto',
+          },
+        }
+);
+
+const DragBlockOverlay = styled('div')(({ isDragging }) => ({
+  display: isDragging ? 'block' : 'none',
+  position: 'absolute',
+  top: '0px',
+  right: '0px',
+  bottom: '0px',
+  left: '0px',
+}));
+
+const defaultSizes = {
+  addonPanel: {
+    down: 200,
+    right: 400,
+  },
+  storiesPanel: {
+    left: 250,
+    top: 400,
+  },
+};
+
+const saveSizes = sizes => {
+  try {
+    localStorage.setItem('panelSizes', JSON.stringify(sizes));
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+const getSavedSizes = sizes => {
+  try {
+    const panelSizes = localStorage.getItem('panelSizes');
+    if (panelSizes) {
+      return JSON.parse(panelSizes);
+    }
+    saveSizes(sizes);
+    return sizes;
+  } catch (e) {
+    saveSizes(sizes);
+    return sizes;
+  }
+};
+
+class Layout extends Component {
+  constructor(props) {
```

Comment:
A decent amount of other components in this PR have their constructor and method bindings replaced by class properties and fields, while a lot have not. Would it be wise to choose 1 and stick with it throughout the project?

Reply:
I tend to write components using modules and module scope function and less with component instance function. That's just my coding style. (composition over inheritance)

Though I do believe that in a perfect world a codebase would be perfectly uniform, in reality code is messy for lots of reasons. I'd rather have people contribute and do a bit of gradual cleanup afterwards.

I personally think that code uniformity is highly overrated. It's very easy to be opinionated about and loose valuable time controlling petty details.

So to answer your question:
Yes it would be wise, but it's not something I'm willing to enforce, because there is a trade-off with getting new developers on board whose opinion and expertise is going to be different. I just apply the boy-scout rules: leave the campsite better then it was.